### PR TITLE
Implement the normalizer of a finite dimensional Lie algebra

### PR DESCRIPTION
**Pull request by tscrim**
We implement the normalizer of a finite dimensional Lie algebra.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/categories/finite_dimensional_lie_algebras_with_basis.py [hunk 1]
```diff
@@ -531,6 +540,99 @@ def center(self):
...
+                sage: S = L.subalgebra([a, e])
+                sage: L.normalizer_basis(S)
+                [a, b, c, e]
+
+            When the subalgebra is the ambient Lie algebra, we return the
+            basis of the ambient Lie algebra::
+
+                sage: L.normalizer_basis(L)
+                Finite family {'a': a, 'b': b, 'c': c, 'd': d, 'e': e}
+                sage: L.normalizer_basis([a,b,c,a,d+e,a+e])
```
Comment (mkoeppe) [inline src/sage/categories/finite_dimensional_lie_algebras_with_basis.py]:
```suggestion
                sage: L.normalizer_basis([a, b, c, a, d + e, a + e])
```

File: src/sage/categories/finite_dimensional_lie_algebras_with_basis.py [hunk 2]
```diff
@@ -531,6 +540,99 @@ def center(self):
...
+                m = matrix([v.to_vector() for v in self.echelon_form(S)])
+            else:
+                m = self.subalgebra(S).basis_matrix()
+
+            if m.nrows() == self.dimension():
+                return self.basis()
+
+            S = self.structure_coefficients()
+            sc = {}
+            for k in S.keys():
```
Comment (fchapoton) [inline src/sage/categories/finite_dimensional_lie_algebras_with_basis.py:591]: do we need `.keys()` here ?

Comment (tscrim) [reply src/sage/categories/finite_dimensional_lie_algebras_with_basis.py:591]: It is a `Family` being returned:
```
sage: L = LieAlgebra(QQ, cartan_type=['A',2])
sage: type(L.structure_coefficients())
<class 'sage.sets.family.FiniteFamily'>
```

**Review by mkoeppe — APPROVED** [review]
LGTM.

**Comment by tscrim** [comment]
Thank you.

**Pull request merged**